### PR TITLE
xftp: delete agent rcv files on completion, error, item delete

### PR DESCRIPTION
**Pull request by spaced4ndy**

### CODE REVIEW

File: src/Simplex/Chat.hs [hunk 1]
```diff
@@ -3753,7 +3771,7 @@ cancelSndFileTransfer user@User {userId} ft@SndFileTransfer {fileId, connId, age
           void . sendDirectMessage conn (BFileChunk sharedMsgId FileChunkCancel) $ ConnectionId connId
         _ -> withAgent $ \a -> void . sendMessage a acId SMP.noMsgFlags $ smpEncode FileChunkCancel
       pure fileConnId
-    fileConnId = if isJust fileInline then Nothing else Just acId
+    fileConnId = if isNothing fileInline then Just acId else Nothing
```
Comment (epoberezkin) [inline src/Simplex/Chat.hs:3774]: not sure if this should check for xftp as well?

Comment (epoberezkin) [reply src/Simplex/Chat.hs:3774]: and not sure why this change is needed at all

File: src/Simplex/Chat/Store.hs [hunk 2]
```diff
@@ -2801,13 +2804,13 @@ updateSndFTDeliveryXFTP db SndFileTransfer {connId, fileId, fileDescrId} msgDeli
...
-getAgentSndFileIdXFTP db User {userId} aSndFileId =
+getSndFileDbIdByAgentId :: DB.Connection -> User -> AgentSndFileId -> ExceptT StoreError IO FileTransferId
+getSndFileDbIdByAgentId db User {userId} aSndFileId =
   ExceptT . firstRow fromOnly (SESndFileNotFoundXFTP aSndFileId) $
     DB.query db "SELECT file_id FROM files WHERE user_id = ? AND agent_snd_file_id = ?" (userId, aSndFileId)
 
-getAgentRcvFileIdXFTP :: DB.Connection -> AgentRcvFileId -> ExceptT StoreError IO FileTransferId
-getAgentRcvFileIdXFTP db aRcvFileId =
+getRcvFileDbIdByAgentId :: DB.Connection -> AgentRcvFileId -> ExceptT StoreError IO FileTransferId
+getRcvFileDbIdByAgentId db aRcvFileId =
```
[the remark this answers is not in the document]
Comment (epoberezkin) [reply src/Simplex/Chat/Store.hs:2813]: maybe just `getRcvFileIdXFTP`?

File: src/Simplex/Chat.hs (hunk 1 above)
Comment (spaced4ndy) [reply src/Simplex/Chat.hs:3774]: that it's not xftp is checked in cancelSndFile, this one is per connection checking XFTP send here would require passing additional data to it

**Pull request merged**